gui/dialogs/dialogoptimize.cpp

```diff
@@ -201,10 +260,29 @@ void DialogOptimize::updatePlots() {
     ui->pltParams->replot(QCustomPlot::RefreshPriority::rpQueuedReplot);
   }
   m_nPlottedIterations = nAvailableIterations;
+
   if (auto img{m_opt->getUpdatedBestResultImage(
           static_cast<std::size_t>(ui->cmbTarget->currentIndex()))};
       img.has_value()) {
     ui->lblResult->setImage(img.value());
+    if (vizMode == VizMode::_2D) {
+      ui->lblResult->setZSlider(ui->zaxisValues);
+      ui->lblResult->setZIndex(ui->zaxisValues->value());
+    }
```

Comment:
could we just do `ui->lblResult->setZSlider(ui->zaxisValues);` once in init() instead? Then I don't think this would be needed - the zslider would remain connected